api/envoy/extensions/credentials/oauth2/v3/oauth2.proto

```diff
@@ -0,0 +1,70 @@
+syntax = "proto3";
+
+package envoy.extensions.credentials.oauth2.v3;
+
+import "envoy/config/core/v3/http_uri.proto";
+import "envoy/extensions/transport_sockets/tls/v3/secret.proto";
+
+import "xds/annotations/v3/status.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.credentials.oauth2.v3";
+option java_outer_classname = "Oauth2Proto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/credentials/oauth2/v3;oauth2v3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#protodoc-title: OAuth2 Credential]
+// [#not-implemented-hide:]
+// [#extension: envoy.credentials.oauth2]
+
+// OAuth2 extension can be used to retrieve an OAuth2 access token from an authorization server and inject it into the
```

Comment:
A few high-level comments:
1. It is unclear why this is different than [forward_bearer_token](https://github.com/envoyproxy/envoy/blob/f32c47e440fd95ae46ed2910a794329238afa886/api/envoy/extensions/filters/http/oauth2/v3/oauth.proto#L111), that IIUC does the oauth and forwards the auth-token. Please clarify what is the different in the description.
2. It is unclear what will be injected if the authentication fails. Please clarify the description.
3. Why is this too different than the oauth2 filter configuration. Is the assumption that the underlying implementation will also be different?

Reply:
As the description of this PR mentioned: This filter is intended to be used for workload authentication, which means that the identity associated with the inserted credential is considered as the identity of the workload behind the envoy proxy(in this case, envoy is typically deployed as a sidecar alongside that workload). Please note that this filter does not handle end user authentication. Its purpose is solely to authenticate the workload itself. 

With regard to the oauth2 extension type: this extension acquires OAuth2 access token through [Client Credentials Grant](https://datatracker.ietf.org/doc/html/rfc6749#section-4.4) flow, which is a purely server-side process.  The oauth2 filter,by contrast, retrieves token through  [Authorization Code Grant](https://datatracker.ietf.org/doc/html/rfc6749#section-4.1) flow which requires user interactions.